addons/point_of_sale/static/src/js/Popups/MoneyDetailsPopup.js

```diff
@@ -0,0 +1,68 @@
+odoo.define('point_of_sale.MoneyDetailsPopup', function(require) {
+    'use strict';
+
+    const { useState, useRef } = owl.hooks;
+    const PosComponent = require('point_of_sale.PosComponent');
+    const Registries = require('point_of_sale.Registries');
+
+    /**
+     * Even if this component has a "confirm and cancel"-like buttons, this should not be an AbstractAwaitablePopup.
+     * We currently cannot show two popups at the same time, what we do is mount this component with its parent
+     * and hide it with some css. The confirm button will just trigger an event to the parent.
+     */
+    class MoneyDetailsPopup extends PosComponent {
+        constructor() {
+            super(...arguments);
+            this.currency = this.env.pos.currency;
+            this.state = useState({
+                moneyDetails: Object.fromEntries(this.env.pos.bills.map(bill => ([bill.value, 0]))),
+                total: 0,
+            });
+        }
+        get firstHalfMoneyDetails() {
+            const moneyDetailsKeys = Object.keys(this.state.moneyDetails).sort((a, b) => a - b);
+            return moneyDetailsKeys.slice(0, moneyDetailsKeys.length/2);
+        }
+        get lastHalfMoneyDetails() {
+            const moneyDetailsKeys = Object.keys(this.state.moneyDetails).sort((a, b) => a - b);
+            return moneyDetailsKeys.slice(moneyDetailsKeys.length/2, moneyDetailsKeys.length);
+        }
+        isClosed() {
+            return this.el.classList.contains('invisible')
+        }
+        openPopup() {
+            this.el.classList.remove('invisible');
+        }
+        updateMoneyDetailsAmount() {
+            this.state.total = Object.entries(this.state.moneyDetails).reduce((total, money) => total + money[0] * money[1], 0);
+        }
+        _closePopup() {
+            this.el.classList.add('invisible');
+        }
+        confirm() {
+            let moneyDetailsNotes = this.state.total  ? 'Money details: \n' : null;
+            this.env.pos.bills.forEach(bill => {
+                if (this.state.moneyDetails[bill.value]) {
+                    moneyDetailsNotes += `  - ${this.state.moneyDetails[bill.value]} x ${this.env.pos.format_currency(bill.value)}\n`;
+                }
+            })
+            const payload = { total: this.state.total, moneyDetailsNotes, moneyDetails: { ...this.state.moneyDetails }};
+            this.trigger('money-details-validated', payload);
+            this._closePopup();
+        }
+        reset() {
+            for (let key in this.state.moneyDetails) { this.state.moneyDetails[key] = 0 }
+            this.state.total = 0;
+        }
```

Comment:
If this is not called anywhere else, I guess you can just merge this with the `discard` method below.

Reply:
It's used in `ClosePosPopup` here:
```js
        openDetailsPopup() {
            if (this.moneyDetailsRef.comp.isClosed()){
                this.moneyDetailsRef.comp.openPopup();
                this.state[this.defaultCashDetails.name].counted = 0;
                if (this.manualInputCashCount) {
                    this.moneyDetailsRef.comp.reset();
                }
            }
        }
```
Basically, we reset the everything from the money details if the input was manually done

Reply:
Oh yeah, I didn't notice that.

Reply:
I've put the same behavior for the opening cash popup, so that everything is coherent